core/src/main/java/com/linecorp/armeria/internal/RequestContextAwareRpcResponse.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.RpcResponse;
+
+final class RequestContextAwareRpcResponse extends RequestContextAwareCompletableFuture<Object>
```

Comment:
All callbacks (e.g `rpcResponse.thenApply(...)`) does not return `RequestContextAwareRpcResponse` but `RequestContextAwareCompletableFuture`.
I think it's right behavior not to return `RequestContextAwareRpcResponse`, but if I was wrong please let me know.